waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -134,17 +144,22 @@ func (wakuM *WakuMetadata) Request(ctx context.Context, peerID peer.ID) (*protoc
 
 	stream.Close()
 
-	if response.ClusterId == nil {
+	if response.GetClusterId() == 0 {
```

Comment:
I *think* that spec is outdated now, since we're assuming in nwaku too that clusterID 0 means no support for metadata protocol. 
cc: @alrevuelta @jm-clius

Reply:
I wouldn't view this as a spec change - just a limited implementation. In other words, so far we have only implemented the Metadata protocol for TWN which is on cluster ID = `1`. This is to avoid breaking backwards compatibility for now, but this does not mean that the protocol fundamentally views `0` as "no support for protocol".